compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs

```diff
@@ -183,9 +185,9 @@ parseReplInput input dflags =
 
 -- | Load all packages in the given session.
 --
--- Returns the list of modules in the main DALFs.
-loadPackages :: ReplClient.Handle -> IdeState -> IO [ImportDecl GhcPs]
-loadPackages replClient ideState = do
+-- Returns the list of modules in the specified import packages.
+loadPackages :: [String] -> ReplClient.Handle -> IdeState -> IO [ImportDecl GhcPs]
```

Comment:
GHC already forbids segments (things separated by `-`) that only consist of numbers https://gitlab.haskell.org/ghc/ghc/-/blob/master/compiler/GHC/Utils/Misc.hs#L1060 so you can uniquely parse things into a name and an optional version and parse in `(PackageName, Maybe PackageVersion)` here (`splitUnitId` does that). Maybe that would be a bit cleaner?

Reply:
:+1: I've pulled the `(PackageName, Maybe PackageVersion)` conversion up into the CLI parser and added validation of package names as well.